packages/store/README.md

```diff
@@ -0,0 +1,223 @@
+# Scalar Collection
+
+A powerful data store for OpenAPI documents that handles `$ref` pointers with Vue reactivity support.
+
+## Features
+
+- Seamless reference resolution
+- Vue reactivity support
+- Circular reference handling
+- Original document structure preservation
+- Caching for resolved references
+
+## Installation
+
+```bash
+npm install @scalar/store
+```
+
+## Basic Usage
+
+Create a new collection instance:
+
+```ts
+import { createCollection } from '@scalar/store'
+
+// Create a collection from an OpenAPI document containing $ref's
+const collection = createCollection({
+  openapi: '3.1.1',
+  info: {
+    title: 'Hello World',
+    version: '1.0.0',
+  },
+  components: {
+    schemas: {
+      Person: {
+        type: 'object',
+        properties: {
+          name: { type: 'string' },
+        },
+      },
+      User: {
+        $ref: '#/components/schemas/Person',
+      },
+    },
+  },
+})
+
+// Access the data without caring about $ref's
+console.log(collection.document.components.schemas.User)
+
+// Output: { type: 'object', properties: { name: { type: 'string' } } }
+```
+
+### Reference Resolution
+
+The store automatically resolves JSON References (`$ref`) when accessing properties:
+
+```ts
+const collection = createCollection({
+  components: {
+    schemas: {
+      Person: {
+        type: 'object',
+        properties: {
+          name: { type: 'string' },
+        },
+      },
+      User: {
+        $ref: '#/components/schemas/Person',
+      },
+    },
+  },
+})
+
+// Both paths access the same data
+const person = collection.document.components.schemas.Person
+const user = collection.document.components.schemas.User
+
+// Changes through either path update the source
+user.properties.age = { type: 'number' }
```

Comment:
okay so we probably don't want to do this, make it immutable and use mutators instead